src/Data/Cursor/PostObjectCursor.php

```diff
@@ -169,13 +169,13 @@ private function compare_with( $by, $order ) {
 				break;
 		}
 
-		$value = $this->cursor_node->{$by} ?? null;
+		$value     = $this->cursor_node->{$by} ?? null;
+		$cursor_id = $this->cursor_node->ID ?? null;
 
 		/**
 		 * Compare by the post field if the key matches a value
 		 */
-		if ( ! empty( $value ) ) {
-
+		if ( ! empty( $value ) && ! empty( $cursor_id ) && ! metadata_exists( 'post', $cursor_id, $by ) ) {
```

Comment:
I might be wrong but I think `metadata_exists()` adds an extra db query.
If so perhaps there's a way to break down this conditional so it's only called when necessary?

Reply:
It should not, because the `cursor_node` is already cached with it's meta. https://github.com/WordPress/wordpress-develop/blob/6.2/src/wp-includes/meta.php#L751